apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3325,6 +3424,18 @@ handle_call_(St, _Req, _From, D) when ?TRANSITION_STATE(St) ->
 handle_call_(_St, _Req, From, D) ->
     keep_state(D, [{reply, From, {error, unknown_request}}]).
 
+handle_info({'DOWN', MRef, process, Client, _} = M,
+            #data{ client = Client, client_mref = MRef } = D) ->
+    lager:debug("Client disconnected ~p", [Client]),
+    case D#data.client_may_disconnect of
```

Comment:
Moving this case clause into a function clause would improve readability.

Reply:
Done